### PR TITLE
MONAI integration and metrics refactoring

### PR DESCRIPTION
**Pull request by SYangster**
- Remove MetricRetriever, change to use MetricRelay
- Change MetricsExchanger to MetricsSender, now can be configured as component
- Unify LogWriter and LogWriterForMetricsExchanger, along with [MLflow, TB, WandB]WriterForMetricsExchanger
- Update monai nvflare integration for monai v1.3.0

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/app_common/widgets/metric_relay.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def __init__(
         heartbeat_timeout=30.0,
         pipe_channel_name=PipeChannelName.METRIC,
         event_type: str = ANALYTIC_EVENT_TYPE,
+        fed_event: bool = True,
```
[the remark this answers is not in the document]
Comment (SYangster) [reply nvflare/app_common/widgets/metric_relay.py:39]: Using the MetricRelay with the client api is configured to fire a federated event.

File: nvflare/app_common/widgets/metric_relay.py [hunk 2]
```diff
@@ -81,7 +83,7 @@ def _pipe_status_cb(self, msg: Message):
     def _pipe_msg_cb(self, msg: Message):
         if not isinstance(msg.data, DXO):
             self.logger.error(f"bad metric data: expect DXO but got {type(msg.data)}")
-        send_analytic_dxo(self, msg.data, self._fl_ctx, self._event_type, fire_fed_event=True)
+        send_analytic_dxo(self, msg.data, self._fl_ctx, self._event_type, fire_fed_event=self._fed_event)
```
[the remark this answers is not in the document]
Comment (SYangster) [reply nvflare/app_common/widgets/metric_relay.py:86]: For the MetricRelay in the monai integration, I set this to false in the spleen_ct_segmentation_local config_fed_client.json

[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/app_common/widgets/metric_relay.py:86]: If this is set to True, then we don't need the component "ConvertToFedEvent".
if this is set to False, then we need to configure "ConvertToFedEvent".

So we can do either way.

File: integration/monai/monai_nvflare/nvflare_stats_handler.py [hunk 3]
```diff
@@ -40,11 +40,11 @@
 DEFAULT_TAG = "Loss"
 
 
-class NVFlareStatsHandler(LogWriterForMetricsExchanger):
+class NVFlareStatsHandler(LogWriter):
```
Comment (YuanTingHsieh) [inline integration/monai/monai_nvflare/nvflare_stats_handler.py:43]: changes in this PR LGTM as a first stage.

We can think of these questions:
- Is `NVFlareStatsHandler` a `LogWriter`? should we inherit it? or should we use some composition?

File: nvflare/app_common/metrics_exchange/metrics_sender.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.utils.analytix_utils import create_analytic_dxo
+from nvflare.app_common.widgets.streaming import AnalyticsSender
+from nvflare.fuel.utils.constants import PipeChannelName
+from nvflare.fuel.utils.pipe.pipe import Message, Pipe
+from nvflare.fuel.utils.pipe.pipe_handler import PipeHandler
+
+
+class MetricsSender(AnalyticsSender):
```
Comment (YuanTingHsieh) [inline nvflare/app_common/metrics_exchange/metrics_sender.py:27]: changes in this PR LGTM as a first stage.

We can think of these questions:
- Is `MetricsSender` a `AnalyticsSender`? should we inherit it? or should we use some composition? or should it be just `FLComponent`? or a `Widget`?

**Pull request merged**